### PR TITLE
PIMS-1542 SRES Deleting/Editing Properties

### PR DESCRIPTION
**Pull request by Fosol**
# Description

We now have a way to identify Roles as *private* or *public*.  Public Roles will be displayed in the **Access Request Page** so that users can select them.

There is a new SRES Role, which includes the `property-delete` and `admin-properties` Claims.  This allows us for now to add the SRES Role manually through Keycloak to any users who require it.

The new Claim `property-delete` is required to delete properties of which belong to an agency the user belongs to.

The new Claim `admin-properties` allows a user to do the following;

- Edit properties in other agencies
- Delete properties in other agencies
- View sensitive properties in other agencies

# Development

- EF Migration Refresh required
- Keycloak import script update (refresh keycloak-db)

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -2,9 +2,15 @@
 * text=auto
 
 # Declare files that will always have LF line endings on checkout.
+text eol=lf
```
Comment (Fosol) [inline .gitattributes:5]: Trying to get Windows from always displaying warnings related to eol

File: auth/keycloak/config/realm-export.json [hunk 2]
```diff
@@ -1933,12 +1933,19 @@
       "clientRole": false,
```
Comment (Fosol) [inline auth/keycloak/config/realm-export.json:1933]: Importing the new realm configuration requires refreshing the container and volume.  See the README.md for the steps.

File: backend/dal/Helpers/Extensions/BuildingExtensions.cs [hunk 3]
```diff
@@ -43,10 +43,11 @@ public static IQueryable<Entity.Building> GenerateQuery(this PimsContext context
             // Check if user has the ability to view sensitive properties.
             var userAgencies = user.GetAgencies();
             var viewSensitive = user.HasPermission(Permissions.SensitiveView);
+            var isAdmin = user.HasPermission(Permissions.AdminProperties);
 
             // Users may only view sensitive properties if they have the `sensitive-view` claim and belong to the owning agency.
             var query = context.Buildings.AsNoTracking().Where(b =>
-                !b.IsSensitive || (viewSensitive && userAgencies.Contains(b.AgencyId)));
+                isAdmin || !b.IsSensitive || (viewSensitive && userAgencies.Contains(b.AgencyId)));
```
Comment (Fosol) [inline backend/dal/Helpers/Extensions/BuildingExtensions.cs:50]: `admin-properties` can view sensitive properties.

File: backend/entities/Role.cs [hunk 4]
```diff
@@ -15,6 +15,12 @@ public class Role : LookupEntity<Guid>
         /// <value></value>
         public string Description { get; set; }
 
+        /// <summary>
+        /// get/set - Whether the role is a public role.
+        /// One which new users can request to join.
+        /// </summary>
+        public bool IsPublic { get; set; } = false;
```
Comment (Fosol) [inline backend/entities/Role.cs:22]: A way to identify public and private Roles.

File: backend/tests/core/DatabaseHelper.cs [hunk 5]
```diff
@@ -50,6 +51,7 @@ public static PimsContext CreatePimsContext(this TestHelper helper, string dbNam
             helper.AddSingleton(user);
             var options = new DbContextOptionsBuilder<PimsContext>()
                 .UseInMemoryDatabase(databaseName: dbName)
+                .ConfigureWarnings(m => m.Ignore(InMemoryEventId.TransactionIgnoredWarning))
```
Comment (Fosol) [inline backend/tests/core/DatabaseHelper.cs:54]: For some reason EF started throwing warnings during unit-testing.  This ignores those.

File: backend/api/Areas/Admin/Controllers/ParcelController.cs [hunk 6]
```diff
@@ -313,7 +313,7 @@ public IActionResult UpdateParcel(int id, [FromBody] Model.ParcelModel model)
         /// <param name="model">The parcel model.</param>
         /// <returns>The parcel who was deleted.</returns>
```
Comment (devinleighsmith) [inline backend/api/Areas/Admin/Controllers/ParcelController.cs:314]: /nit the parcel who was deleted

**Comment by devinleighsmith** [comment]
@Fosol have you tested the add parcel API against this?

**Comment by Fosol** [comment]
Add and Update have been tested via Postman.

File: backend/dal/Services/Concrete/BuildingService.cs [hunk 7]
```diff
@@ -44,14 +44,15 @@ public IEnumerable<Building> Get(double neLat, double neLong, double swLat, doub
...
+            var isAdmin = this.User.HasPermission(Permissions.AdminProperties);
 
             IQueryable<Building> query = null;
             // Users may only view sensitive properties if they have the `sensitive-view` claim and belong to the owning agency.
             query = this.Context.Buildings
                 .Include(b => b.Parcel)
                 .AsNoTracking()
                 .Where(p =>
-                (!p.IsSensitive || (viewSensitive && userAgencies.Contains(p.AgencyId))) &&
+                (isAdmin || !p.IsSensitive || (viewSensitive && userAgencies.Contains(p.AgencyId))) &&
```
Comment (asanchezr) [inline backend/dal/Services/Concrete/BuildingService.cs:55]: Doesn't have to happen in this PR but would love to see if you can find a way to extract the `WhoCanSee` functionality into an extension method or utils class or something to delegate this decision. 

Currently, we have to modify the code in multiple places (BuildingService, ParcelService, ParcelExtensions, etc etc) when we want to change/tweak who can see sensitive properties. 

The probability of bugs grows exponentially when we have to resort to copy/paste around the codebase for changes in logic.

**Pull request merged**